File: 2162.js
var input = require('fs').readFileSync('/dev/stdin', 'utf8');
var lines = input.split('\n');
const A = lines.shift();
const picos = lines.shift().split(" ");
let temp;
let below = 0;
let above = 0;
let equal = 0;

for(let i = 0; i < picos.length, i <= parseInt(A); i++){
  if(i == 0){
    temp = parseInt(picos[0]);
  }
  else{
    if(temp < picos[i]){
      temp = parseInt(picos[i]);
      above++;
      below = 0;
    }
    else if(temp > picos[i]){
      temp = parseInt(picos[i]);
      below++;
      above = 0;
    }
    else if(temp == picos[i]){
      equal++;
    }

    if(above == 2 || below == 2 || equal == 1){
      console.log("0");
      equal++
      break;
    }
  }
}


if(above < 2  && below < 2 && equal == 0){
  console.log("1");
}

File: java/2162.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        
        int A = sc.nextInt();
        sc.nextLine(); 
        String[] picos = sc.nextLine().split(" ");
        
        int temp = 0;
        int below = 0;
        int above = 0;
        int equal = 0;
        boolean valid = true;
        
        for (int i = 0; i <= A && i < picos.length; i++) {
            int current = Integer.parseInt(picos[i]);
            
            if (i == 0) {
                temp = current;
            } else {
                if (temp < current) {
                    temp = current;
                    above++;
                    below = 0;
                    equal = 0;
                } else if (temp > current) {
                    temp = current;
                    below++;
                    above = 0;
                    equal = 0;
                } else {
                    equal++;
                }
                
                if (above == 2 || below == 2 || equal == 1) {
                    System.out.println("0");
                    valid = false;
                    break;
                }
            }
        }
        
        if (valid) {
            System.out.println("1");
        }
        
        sc.close();
    }
}
